### ISSUE TITLE
Automatic conversion to LaTeX source

### ISSUE THREAD

**Issue by berceanu** [manubot/rootstock#249]
For some research fields, like math or theoretical physics, one must submit the LaTeX source of the manuscript for publication in, eg. [APS journals](https://journals.aps.org/). It would therefore be very convenient if manubot could also have a LaTeX + BibTeX conversion mode for such cases. iirc `pandoc` supports conversion to TeX?

**Comment by agitter** [comment]
Another benefit of supporting LaTeX could be enabling Manubot-based writing of documents that have precise formatting requirements, like grant applications or university dissertations.  I haven't tested this so it's unclear to me how much the pandoc tex template helps with that or whether the final formatting steps would have to be manual after the content is finalized.  Perhaps this is the same idea as "branded PDFs" above.

**Comment by slochower** [comment]
@agitter agreed, although I found the `pandoc` template system cumbersome. See this existing list: https://github.com/jgm/pandoc/wiki/User-contributed-templates

**Comment by dhimmel** [comment]
@slochower for the `BUILD_PDF_VIA_LATEX` step, would it be possible to take the output `.tex` file from the earlier `pandoc --to=latex` command and pass it directly to the latex compiler? Is there any benefit to running pandoc twice? I was envisioning that once we had a `.tex` and `.bib` file, we would no longer need to use pandoc.

**Comment by slochower** [comment]
I think using the output of `pandoc --to=latex` should work, modulo the figure paths. I _think_ when I did this earlier, I specified the figures in the current path, e.g., `[Caption.](figure.png)` in the Markdown (as usual). If you covert this to LaTeX (without having `pandoc` make the PDF for you), I think you'll need to either symlink the figures to the `.tex` directory or vice versa.

**Comment by dhimmel** [comment]
Here's a [useful resource](https://github.com/PHPirates/travis-ci-latex-pdf) on different ways to install LaTeX on Travis CI. It mentions [`tectonic`](https://tectonic-typesetting.github.io/en-US/), which seems to be a more user-friendly `xelatex` (although I don't have a good understanding of how all the LaTeX infrastructure fits together).

**Comment by dhimmel** [comment]
I've got the `pandoc --to=pdf --pdf-engine=xelatex` workflow to produce a PDF. However, I'd like to see what Pandoc does to generate the LaTeX, so that we can potentially replicate it in an `output/latex` directory that could contain a standalone LaTeX source. Pandoc creates a temporary directory as part of the LaTeX processing, but there is no builtin way to retain that directory (see https://github.com/jgm/pandoc/issues/2288).

Setting `--pdf-engine-opt=-output-directory=output/latex` did write some files including the pdf to `output/latex/input.*`, but then Pandoc erred with `Error producing PDF.`. Possible related discussion at https://github.com/jgm/pandoc/issues/4721.

**Comment by habi** [comment]
>    Should we also output a `.bib` file with the reference metadata, or should this be included in the .tex?

Usually, from what I've seen submission systems that can ingest `.tex` files require the bibliography to be included in the file, so I'd vote for the latter.

**Comment by habi** [comment]
I'd also really like to have a LaTeX file as output, so I don't have to fiddle with `pandoc` myself.
I'm happy to do some more manual work with it (e.g. applying a template myself), but having a .tex file *with* bibliography would speed up the process greatly.

If `manubot` generates a .tex file somewhere, then https://github.com/xu-cheng/latex-action might be of help, which [I've used](https://github.com/habi/latex-test) to compile such a .tex to a PDF with GitHub Actions.

**Comment by dhimmel** [comment]
> having a .tex file with bibliography would speed up the process greatly.

@habi I propose the simplest possible LaTeX export in https://github.com/manubot/rootstock/pull/384. In https://github.com/manubot/rootstock/pull/256, I tried to get the LaTeX to compile and render as PDF, which proved challenging. But perhaps having a `.tex` file will help you to a sufficient extent. So please check our #384 and let us know whether it works for your application.

**Comment by agitter** [comment]
I'm sharing some notes from our Manubot manuscript that we exported to LaTeX for a conference submission.  There are more details at https://github.com/greenelab/covid19-review/pull/943.

We customized a LaTeX template for the conference style.  Like @slochower, we also found the templating system cumbersome.  The Manubot metadata didn't perfectly fit the template expectations so we had a bidirectional process of modifying the template and the metadata.  Getting the authors to show up correctly was the trickiest part.  We created a new metadata.yaml file using a Python script, in part because we were already modifying metadata.yaml programmatically because this conference submission was one piece of a larger project.

It's very helpful that newer versions of pandoc can convert the CSL JSON file Manubot produces into a .bib file.  I typically submit a .bib file to a conference or journal instead of embedding references in a .tex file.  We added this conversion step to the build script.  We used a regex to strip out the `note` fields from the .bib file.  We also used custom pandoc settings in a new yaml file, including `cite-method: natbib` for the references.

We didn't try to build a PDF with continuous integration.  We used Manubot to get 95% of the way to submission automatically and then fine-tuned LaTeX issues in Overleaf before submitting.
